File: 0169-majority-element/0169-majority-element.cpp
#include <vector>
#include <unordered_set>

class Solution {
public:
    int majorityElement(vector<int>& nums) {
    int freq = nums.size()/2;
    unordered_set<int> cpy(nums.begin(), nums.end());
    for(int i: cpy) {
        if(count(nums.begin(), nums.end(), i) > freq) {
            return i; 
        }
    }
    
    return -1;
    }
};

File: 0169-majority-element/0169-majority-element.cs
public class Solution {
    public int MajorityElement(int[] nums) {
        if(nums.Length == 1)
            return nums[0];
        int freq = nums.Length/2;
        Dictionary<int, int> myDic = new Dictionary<int, int>();
        for(int i=0; i<nums.Length; i++)
        {
            if(!myDic.TryAdd(nums[i], 1))
            {
                myDic[nums[i]] = myDic[nums[i]]+1;
                if(myDic[nums[i]] > freq)
                    return nums[i];
            }
        }
        return -1;

    }
}